### PR TITLE
docs(button): document accessibility concerns

### PR DESCRIPTION
**Pull request by hchevallier**
# General summary

Adds some documentation about accessibility for the `<Button>` component

# Screenshots

### CODE REVIEW

File: packages/site-demo/content/product/components/button/index.mdx [hunk 1]
```diff
@@ -40,6 +40,26 @@ Use full width when you need to get the button full width.
...
+You **should define the `alt` attribute**, with a string indicating where the link will take the user.
+- **Toggle button**, when your button as an active/inactive state (`isSelected` prop is responsible of this state). 
+If the label of the button **changes depending on the state, then it doesn't need anything more.**
+But if the label of the button **doesn't change you should add a `aria-pressed=true/false`** to specify to the user in which state the button is.
+ - **Menu button** if your button opens a dropdown. 
+ You have to set, on the button, `aria-haspopup` to `true`, and `aria-expanded` to `true` when the dropdown is open.
+
+#### IconButton
+
+All concerns about the `Button` component are true for `IconButton`, but as there is no text in the button, you have to provide an `aria-label` describing what the button does.
```
Comment (gcornut) [inline packages/site-demo/content/product/components/button/index.mdx:61]:
```suggestion
All concerns about the `Button` component are true for `IconButton`, but as there is no text in the button, you have to provide a `label` describing what the button does.
```

**Review by gcornut — CHANGES_REQUESTED** [review]
Can we document the props `aria-pressed`, `aria-haspopup` and `aria-expanded` in the base button props JSDoc?

File: packages/site-demo/content/product/components/button/index.mdx [hunk 2]
```diff
@@ -40,6 +40,26 @@ Use full width when you need to get the button full width.
...
+
+Buttons may have 3 specific accessibility use cases:
+
+- **Button as a link** if you set the `linkAs` or the `href` prop, then your button will use the `<a>` tag.
+You **should define the `alt` attribute**, with a string indicating where the link will take the user.
+- **Toggle button**, when your button as an active/inactive state (`isSelected` prop is responsible of this state). 
+If the label of the button **changes depending on the state, then it doesn't need anything more.**
+But if the label of the button **doesn't change you should add a `aria-pressed=true/false`** to specify to the user in which state the button is.
+ - **Menu button** if your button opens a dropdown. 
+ You have to set, on the button, `aria-haspopup` to `true`, and `aria-expanded` to `true` when the dropdown is open.
```
Comment (gcornut) [inline packages/site-demo/content/product/components/button/index.mdx:57]: "Menu" is a very specific ARIA pattern, should we be more general and describe the popup use case (menu, combobox, dialog, etc.) and separately describe the expanded use case (menu, disclosure, popup, etc.) ?
Also it looks like `aria-expanded` should always be paired with `aria-controls` or `aria-owns`

**Review by gcornut — CHANGES_REQUESTED** [review]

File: packages/site-demo/content/product/components/button/index.mdx (hunk 2 above)
Comment (hchevallier) [reply packages/site-demo/content/product/components/button/index.mdx:57]: I took the [`Action Menu Button`](https://w3c.github.io/aria-practices/examples/menu-button/menu-button-actions.html) as a reference, this use case is listed in the [Button spec](https://w3c.github.io/aria-practices/#button)
I'm not sure what you mean, for dialog pattern I think it should be specified in the Dialog component. Maybe we can add here some links to the different DS components docs, to list all specific patterns 

`aria-controls` can't refer to an element that is not in the DOM, not sure this is relevant in our case, but I may be wrong

Comment (hchevallier) [reply packages/site-demo/content/product/components/button/index.mdx:57]: Ok just read that: 
> The aria-controls only needs to be set when the popup is visible, but it is valid and easier to program to reference an element that is not visible.
I'll add this

Comment (hchevallier) [reply packages/site-demo/content/product/components/button/index.mdx:57]: > Also, when the dropdown is open, you have to add `aria-controls` on the button, referencing the id of the dropdown. 
 If your dropdown is not present in the DOM when it is closed, then you should not provide the `aria-controls` prop.

 Does it looks ok for you?

Comment (gcornut) [reply packages/site-demo/content/product/components/button/index.mdx:57]: Yep, it should be sufficient

File: packages/site-demo/content/product/components/button/index.mdx [hunk 3]
```diff
@@ -40,6 +40,27 @@ Use full width when you need to get the button full width.
...
+Buttons may have 3 specific accessibility use cases:
+
+- **Button as a link** if you set the `linkAs` or the `href` prop, then your button will use the `<a>` tag.
+In this case, the label of the button (or `aria-label`) will be enough to make your link accessible.
+- **Toggle button**, when your button as an active/inactive state (`isSelected` prop is responsible of this state). 
+If the label of the button **changes depending on the state, then it doesn't need anything more.**
+But if the label of the button **doesn't change you should add a `aria-pressed=true/false`** to specify to the user in which state the button is.
+ - **Menu button** if your button opens a dropdown. 
+ You have to set, on the button, `aria-haspopup` to `true`, and `aria-expanded` to `true` when the dropdown is open.
+ Also, when the dropdown is open, you have to add `aria-controls` on the button, referencing the id of the dropdown. If your dropdown is not present in the DOM when it is closed, then you should not provide the `aria-controls` prop.
```
Comment (gcornut) [inline packages/site-demo/content/product/components/button/index.mdx:58]: We'll look into making such a pattern trivial with a new Menu/MenuTrigger component

**Review by gcornut — APPROVED** [review]

File: packages/site-demo/content/product/components/button/index.mdx (hunk 3 above)
Comment (hchevallier) [reply packages/site-demo/content/product/components/button/index.mdx:58]: I agree

**Pull request merged**